shared/src/commonMain/kotlin/at/asitplus/wallet/app/common/Extensions.kt

```diff
@@ -52,53 +51,58 @@ fun InputDescriptor.extractConsentData(): Triple<CredentialRepresentation, Const
         this.format?.msoMdoc != null -> ISO_MDOC
         else -> PLAIN_JWT
     }
-    val credentialIdentifiers: Collection<String> = when (credentialRepresentation) {
+    val scheme = when (credentialRepresentation) {
         PLAIN_JWT -> throw Throwable("PLAIN_JWT not implemented")
-        SD_JWT -> vctConstraint()?.filter?.referenceValues()
-        ISO_MDOC -> listOf(this.id)
-    } ?: throw Throwable("Missing Pattern")
+        SD_JWT -> vctConstraint()?.filter?.referenceValues()?.let { values ->
+            values.firstNotNullOfOrNull { AttributeIndex.resolveSdJwtAttributeType(it) }
+                ?: FallbackCredentialScheme(sdJwtType = values.first())
+        }
 
-    val scheme = AttributeIndex.schemeSet.firstOrNull {
-        it.matchAgainstIdentifier(credentialRepresentation, credentialIdentifiers)
-    } ?: throw Throwable("Missing scheme for $credentialIdentifiers")
+        ISO_MDOC -> AttributeIndex.resolveIsoDoctype(id)
+            ?: FallbackCredentialScheme(isoDocType = id, isoNamespace = id)
+    } ?: FallbackCredentialScheme()
 
-    val matchedCredentialIdentifier = when (credentialRepresentation) {
-        PLAIN_JWT -> throw Throwable("PLAIN_JWT not implemented")
-        SD_JWT -> if (scheme.sdJwtType in credentialIdentifiers) scheme.sdJwtType else scheme.isoNamespace
-        ISO_MDOC -> scheme.isoDocType
+    val attributes: Map<NormalizedJsonPath, Boolean> = this.constraints?.fields
+        ?.mapNotNull { it.toNormalizedJsonPath() }
+        ?.filterNot { (it.segments.firstOrNull() as? NameSegment)?.memberName?.equals("vct") == true }
+        ?.associateWith { true } // all claims are optional
+        ?: mapOf()
+
+    return Triple(credentialRepresentation, scheme, attributes)
+}
+
+// TODO Replace with function from JSONPath
+private fun ConstraintField.toNormalizedJsonPath(): NormalizedJsonPath? =
```

Comment:
@acrusage-iaik That's obviously not how it's supposed to be, but it kinda "works" (for the limited use case of claim names). Could you sketch a more general function, i.e. from string to NormalizedJsonPath please?

Reply:
That sounds fine!